hms-app/src/components/HackathonDetails.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react'
+import { getHackathonDetails } from '../actions/GetBackendData'
+import IdeaCardList from './IdeaCardList'
+
+type IProps = {
+  hackathonID: string
+}
+
+export default function HackathonDetails(props: IProps) {
+  const { hackathonID } = props
+  const [hackathonData, setHackathonData] = useState({
+    errorhackathonData: false,
+    isLoadinghackathonData: true,
+    title: 'string',
+    startDate: '',
+    endDate: '',
+    participantIds: [],
+    categoryIds: [],
```

Comment:
Since we now return the `XxxPreviewResponse` for everything, these names are probably a bit more accurate:
```suggestion
    participants: [],
    categories: [],
```

Reply:
all of this data could be defined in a single type in your `types.tsx` file too